languages/elm/reference/implementing-a-concept-exercise.md

```diff
@@ -1,5 +1,119 @@
+
+Before implementing the exercise, please make sure you have a good understanding of what the exercise should be teaching (and what not). This information can be found in the exercise's GitHub issue.
+
+Any concept exercise in any v3 track requires the following files to be created:
+
+<pre>
+languages
+└── elm
+    └── exercises
+        └── concept
+            └── &lt;SLUG&gt;
+                ├── .docs
+                |   ├── instructions.md
+                |   ├── introduction.md
+                |   ├── hints.md
+                |   └── after.md (optional)
+                └── .meta
+                |   └── config.json
+                |   └── design.md
+                |   └── Example.elm
+                ├── elm.json
+                ├── src
+                |   └── &lt;NAME&gt;.elm
+                └── tests
+                |   └── &lt;NAME&gt;.elm
+</pre>
+
+## Step 1: add .docs/introduction.md
+
+This file contains an introduction to the concept. It should be explicit about what the exercise teaches and maybe provide a brief introduction to the concepts, but not give away so much that the user doesn't have to do any work to solve the exercise.
+
+## Step 2: add .docs/instructions.md
+
+This file contains instructions for the exercise. It should explicitly explain what the user needs to do (define a method with the signature `X(...)` that takes an A and returns a Z), and provide at least one example usage of that function. If there are multiple tasks within the exercise, it should provide an example of each.
+
+## Step 3: add .docs/hints.md
+
+If the user gets stuck, we will allow them to click a button requesting a hint, which shows this file. We will softly discourage them using it. The file should contain both general and task-specific "hints". These hints should be enough to unblock almost any student.
+
+## Step 4: add .docs/after.md (optional)
+
+Once the user completes the exercise they will be shown this file, which recaps the learning goals of the exercise, and secondarily gives them any bonus information or further reading about the concept taught.
+
+## Step 5: update languages/&lt;TRACK&gt;/config.json
+
+An entry should be added to the track's `config.json` file for the new concept exercise:
+
+```json
+{
+  ...
+  "exercises": {
+    "concept": [
+      ...
+      {
+        "slug": "<SLUG>",
+        "uuid": "<UUID>",
+        "concepts": ["<CONCEPT-1>"],
+        "prerequisites": ["<PREREQUISITE-1>", "<PREREQUISITE-2>"]
+      }
+    ]
+  }
+}
+```
+
+## Step 6: adding track-specific files
+
+Having added the files that are not specific to the track, now is the time to create the track-specific files. These file will include:
+
+- An elm.json file.
+- A stub implementation file (src/&lt;NAME&gt;.elm).
+- A file containing the test suite (tests/&lt;NAME&gt;.elm).
+- An example implementation file that passes all the tests (.meta/Example.elm).
```

Comment:
In general, we try to quote file names (we might not do this consistently though):

```suggestion
- An `elm.json` file.
- A stub implementation file (`src/<NAME>.elm`).
- A file containing the test suite (`tests/<NAME>.elm`).
- An example implementation file that passes all the tests (`.meta/Example.elm`).
```